### ISSUE TITLE
Optimizer may not be necessary in functions "targeted_detection" and "untargeted_detection" ?

### ISSUE THREAD

**Issue by ywu36** [s-huu/TurningWeaknessIntoStrength#6]
Dear author,

In functions "targeted_detection" and "untargeted_detection", a SGD optimizer is initialized at the very beginning, but it seems that no "optimizer_s.step()" is run inside of the functions. May I know the reason why we need the "optimizer_s"? It seems that "loss.backward()" and "x_var.grad.data" can do the job without the "optimizer_s". Thanks. 

Best regards,
YH

**Comment by s-huu** [comment]
Hi.

Yeah I agree with you that the optimizer here is not necessary. Thanks for pointing that out. We will make the fix!
